RFC/0001-graph-display.md

```diff
@@ -0,0 +1,241 @@
+
+- Add new nodes of any type
+- Drag nodes to a new position
+- Draw connections between nodes with the mouse
+- Delete connections
+- Auto-arrange nodes
+- Display the nodes at multiple zoom levels
+- Scroll around the node environment
+
+None of these actions should leave the graph in an invalid state.
+
+## 5 Drawbacks
+
+- This project provides a significant time sink for at least one developer.
+
+- Even once the general library is written, more developer time will
+  be needed to write the Dissolve specific uses of the library
+
+
+## 6 Alternatives
+
+[Qt Node Editor](https://github.com/paceholder/nodeeditor) does not
+natively support QML and thus will require significant work to provide
+a consistent experience with the rest of the application.
+
+[QuickQanava](https://github.com/cneben/QuickQanava?tab=readme-ov-file)
+requires the construction and maintenance of an explicit graph object,
+as opposed to providing an abstract interface that we can build our
+own model again.  Additionally, at the time of this writing,
+QuickQanava does not build on Mac or Linux
+
+## 7 Potential Impact and Dependencies
+
+This could have a significant impact on the structure of the input
+file, since it represents a major change in the state of the data.
+Furthermore, this may be a significant break change, as it would not
+be a trivial matter to convert existing, linear code into the new
+graph based setup.
+
+## 8 Unresolved questions
+  
+### 8.1 Invalid Graphs
+
+Users might attempt to connect nodes in incompatible way
+(e.g. connecting a string to a value expecting an int) or fail to
+provide a node with all of the information that it requires.  More
+generally, the system needs to deal with the possibility of invalid
+graphs.
+  
+This first method is to simply forbid invalid graphs.  The model
+should reject any attempt to create invalid edges or delete nodes that
+are required.  This had advantages in design and implementation, but
+places limits on what can provide a good user experience.  For
+example, every possible node input must have a **valid** default
+value, so that the connection to that input can be deleted.
+  
+The second method is to allow invalid graphs, but refuse to run them.
+This provides more flexibility in the user interface, but also
+provides the challenge of informing the user about invalid graphs.  A
+user might edit a graph, then switch to a different tab, come back ten
+hours later and be very confused why their simulation will not run.
+This becomes especially perilous in the case of saving files.  Should
+it possible to save an input file with an invalid graph?  If it is,
+then the user has been handed a tool to create files that break the
+CLI version of Dissolve.  If saving invalid graphs is forbidden, then
+a user is stuck unable to save hours of work until they have fixed the
+problem with their graph.
+
+### 8.2 Underlying iteration model
+
+
+Do we truly want an `AbstractListModel` for the `edges` and `nodes`?
```

Comment:
Probably we do, at least based on how I'm understanding your proposal here as it stands. Creating this as a light framework feels like the best option, with the specific functionality we need only.